# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1316434530 14400
#      Mon Sep 19 08:15:30 2011 -0400
# Branch stable
# Node ID 6b6d0e51bd2f3f23d796ab9291bd672ca4cab724
# Parent  6abc581bf2c9dfba2408981ff54511d1685c34d5
scripts/general/interpn.m: fix(m)==m is faster than round(m)==m

diff --git a/scripts/general/interpn.m b/scripts/general/interpn.m
--- a/scripts/general/interpn.m
+++ b/scripts/general/interpn.m
@@ -87,17 +87,17 @@ function vi = interpn (varargin)
   endif
 
   if (nargs < 3)
     v = varargin{1};
     m = 1;
     if (nargs == 2)
       if (ischar (varargin{2}))
         method = varargin{2};
-      elseif (isnumeric (m) && isscalar (m) && round (m) == m)
+      elseif (isnumeric (m) && isscalar (m) && fix (m) == m)
         m = varargin{2};
       else
         print_usage ();
       endif
     endif
     sz = size (v);
     nd = ndims (v);
     x = cell (1, nd);
